Bad mseed test case

Just stumbled over a bad mseed test case in `obspy/io/mseed/tests/test_mseed_special_issues.py`:

``` python
    def test_wrong_record_length_as_argument(self):
        """ 
        Specifying a wrong record length should raise an error.
        """
        file = os.path.join(self.path, 'data', 'libmseed',
                            'float32_Float32_bigEndian.mseed')
        self.assertRaises(Exception, read, file, reclen=4096)
```

The exception is raised because the file is in data/encoding and the path is wrong, so the whole test case is bogus.. should be fixed.. ;-)

If I put the right path the following exception is raised which is also pretty strange because the file is there and can be opened..

```
                # XXX: Might cause problems if the data is faulty and the user
                # set start/end time. Not sure what to do in this case.
                elif not starttime and not endtime:
>                   raise Exception("Cannot open file/files: %s" % pathname)
E                   Exception: Cannot open file/files: /home/megies/git/obspy-maintenance_1.0.x/obspy/io/mseed/tests/data/encoding/float32_Float32_bigEndian.mseed

../../../core/stream.py:243: Exception
```

--- a/obspy/io/mseed/tests/test_mseed_special_issues.py
+++ b/obspy/io/mseed/tests/test_mseed_special_issues.py
@@ -276,9 +276,18 @@ class MSEEDSpecialIssueTestCase(unittest.TestCase):
         """
         Specifying a wrong record length should raise an error.
         """
-        file = os.path.join(self.path, 'data', 'libmseed',
+        file = os.path.join(self.path, 'data', 'encoding',
                             'float32_Float32_bigEndian.mseed')
-        self.assertRaises(Exception, read, file, reclen=4096)
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter('always')
+            # invalid reclen
+            read(file, reclen=111)
+            self.assertTrue('Invalid record length' in str(w[0].message))
+            self.assertEqual(w[0].category, UserWarning)
+            # wrong reclen - raises and also displays a warning
+            self.assertRaises(Exception, read, file, reclen=4096)
+            self.assertTrue('reclen exceeds buflen' in str(w[1].message))
+            self.assertEqual(w[1].category, InternalMSEEDReadingWarning)
 
     def test_read_with_missing_blockette010(self):
         """

